sympy/matrices/matrices.py

```diff
@@ -3400,7 +3401,10 @@ def diagonalize(self, reals_only=False, sort=False, normalize=False):
                     vec = vects[k]
                     if normalize:
                         vec = vec / vec.norm()
-                    P = P.col_insert(P.cols, vec)
+                    L = Matrix(P)
+                    k1 = P.cols
+                    L.col_insert(k1, vec)
+                    P = ImmutableMatrix(L)
```

Comment:
Though i did not mention this at all. But since the operation of `row_insert()` and `col_insert()` are to be in-place how can they act for `Immutable Matrix` ? . Since `P` is `ImmutableMatrix` i converted it into `Mutable` matrix for `P.col_insert()` is not valid.
```
>>> M = ImmutableMatrix([ [1, 2, 3], [2, 3, 4] ])
>>> M1 = Matrix(M)
>>> V = Matrix([ [10, 10, 10 ] ])
>>> M.row_insert(1, V)
```